internal/repositories/equipment.go

```diff
@@ -185,32 +185,61 @@ func (r *equipmentRepository) DeleteEquipmentPhoto(ctx context.Context, id strin
 	return nil
 }
 
-func (r *equipmentRepository) AllEquipments(ctx context.Context, limit, offset int, orderBy, orderColumn string) ([]*ent.Equipment, error) {
+func (r *equipmentRepository) AllEquipments(
+	ctx context.Context,
+	limit, offset int,
+	orderBy, orderColumn string,
+	includeArchived bool,
+) ([]*ent.Equipment, error) {
 	if !utils.IsValueInList(orderColumn, fieldsToOrderEquipments) {
 		return nil, errors.New("wrong column to order by")
 	}
-	orderFunc, err := utils.GetOrderFunc(orderBy, orderColumn)
+
+	tx, err := middlewares.TxFromContext(ctx)
 	if err != nil {
 		return nil, err
 	}
-	tx, err := middlewares.TxFromContext(ctx)
+
+	orderFunc, err := utils.GetOrderFunc(orderBy, orderColumn)
 	if err != nil {
 		return nil, err
 	}
-	result, err := tx.Equipment.Query().Order(orderFunc).Limit(limit).Offset(offset).
-		WithCategory().WithSubcategory().WithCurrentStatus().WithPetKinds().
-		WithPetSize().WithPhoto().
+
+	query := tx.Equipment.Query().
```

Comment:
What is we make ordering work done by Ent, not ordering manually?

Reply:
There are ordering done by ent and event inclusion and exclusion of values, hence there are an additional double check on sorting function that consists ordering